src/components/GameBoard.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import { Square, Row } from '../constants';
+import Game from '../tic-tac-toe';
+
+class GameBoard extends React.Component<any, any> {
+  constructor(props: {}) {
+    super(props);
+    this.state = {
+      board: [
+        [undefined, undefined, undefined],
+        [undefined, undefined, undefined],
+        [undefined, undefined, undefined],
+      ],
+      boardSize: 0,
+      gameStatus: null,
+    };
+  }
+
+  public handleChange = (event: React.FormEvent<HTMLInputElement>): void => {
+    this.setState({
+      boardSize: event.currentTarget.value,
+    });
+  };
+
+  public submitBoardSize = (event: any): void => {
+    event.preventDefault();
+    const newGame = new Game(Number(this.state.boardSize));
+    console.log('NEW GAME', newGame);
+    this.setState({
+      board: newGame.board,
+      gameStatus: 'In progress',
+    });
+  };
+
+  public render() {
+    const { board } = this.state;
+    return (
+      <div className='container center'>
+        <div className='center-container center column'>
+          <h1>Tic Tac Toe</h1>
+          <p>What size should the board be?</p>
+          <form onSubmit={this.submitBoardSize}>
+            Board Size:{' '}
+            <input onChange={this.handleChange} type='text' name='size' />
+            <input type='submit' value='submit' />
+          </form>
+          {this.state.gameStatus ? (
+            <table>
+              <tbody>
+                {board.map((row: Row, _idx: number) => (
+                  <tr key={_idx}>
+                    {row.map((square: Square, _idx: number) => (
+                      <td key={_idx}>{square || '•'}</td>
```

Comment:
This is masking the outer variables index, also.

Reply:
What would you recommend to be the `key` here? I usually make it whatever unique value I can grab from the data being mapped over, but there's nothing unique here.

Reply:
I was overly prescriptive here in the same way that react warnings are: this is a case where index-as-key is a suitable solution, so long as elements are guaranteed to be generated in the same order (which I think that they are). In that case though, index-as-key is the default behavior, so you might as well not specify them at all. There's great documentation on this [here](https://reactjs.org/docs/reconciliation.html#recursing-on-children) if you're interested.